src/hotspot/share/opto/macro.cpp

```diff
@@ -1631,7 +1630,7 @@ void PhaseMacroExpand::expand_initialize_membar(AllocateNode* alloc, InitializeN
 
 void PhaseMacroExpand::expand_dtrace_alloc_probe(AllocateNode* alloc, Node* oop,
                                                 Node*& ctrl, Node*& rawmem) {
-  if (C->env()->dtrace_extended_probes()) {
+  if (C->env()->dtrace_alloc_probes()) {
```

Comment:
IIUC, this probe is related to object allocation, so it should be expanded when ciEnv::dtrace_alloc_probes() returns true, just like the implementation in C1(see C1_MacroAssembler::initialize_object().

Reply:
Originally it checked `ExtendedDTraceProbes` flag here. But I think it was missing change when `DTraceAllocProbes` flag was introduced by
https://bugs.openjdk.java.net/browse/JDK-6346964

So I agree with this change.